lib/Destiny/Controllers/ProfileController.php

```diff
@@ -467,10 +467,14 @@ public function removeAuthProfiles(array $params): string {
         $userId = Session::getCredentials()->getUserId();
         FilterParams::isArray($params, 'selected');
         $userAuthService = UserAuthService::instance();
+        $currentAuthProvider = Session::getCredentials()->getAuthProvider();
+        $error = false;
         foreach ($params['selected'] as $id) {
-            $userAuthService->removeByIdAndUserId((int) $id, $userId);
+            if($currentAuthProvider !== $id) $userAuthService->removeByIdAndUserId((int) $id, $userId);

```

Comment:
This is always `true` because `$currentAuthProvider` is a string and `$id` is an integer. To fix this, you can modify the frontend such that it passes the names of the providers to delete to this endpoint rather than the IDs, or resolve the name of the current provider to an ID before doing the comparison.

Reply:
Ah, pardon for missing that.

Feel free to close this and add it later however is the easiest way for you as I don't know if it would be easier to add it as a URL parameter to individually remove AUTH profiles or pass the data as an array.